config/confighttp/confighttp.go

```diff
@@ -339,7 +340,11 @@ type CORSSettings struct {
 
 func authInterceptor(next http.Handler, authenticate configauth.AuthenticateFunc) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		ctx, err := authenticate(r.Context(), r.Header)
+		headers := r.Header.Clone()
```

Comment:
Should we remove the same logic that was added as part of the previous PR? Should we do something similar for gRPC here as well?

Reply:
1) I reverted all pervious PR - https://github.com/open-telemetry/opentelemetry-collector/pull/4736
2) For me - no, because this problem only for http. If you want, i can add

Reply:
Instead of reverting that commit, use this PR here alone to show the differences. I believe it's about just moving the logic to a different place. And yes, please fix it for gRPC as well.